modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala

```diff
@@ -111,6 +149,39 @@ package object catseffect {
     pureconfig.saveConfigAsPropertyFile(conf, outputPath, overrideOutputPath, options)
   }
 
+  /**
+   * Save the given configuration into a property file
+   *
+   * @param conf The configuration to save
+   * @param outputPath Where to write the configuration
+   * @param blocker the blocking context which will be used to load the configuration.
+   * @param overrideOutputPath Override the path if it already exists
+   * @param options the config rendering options
+   * @return The return action will save out the supplied configuration upon invocation
+   */
+  def blockingSaveConfigAsPropertyFileF[F[_], A](
+    conf: A,
+    outputPath: Path,
+    blocker: Blocker,
+    overrideOutputPath: Boolean = false,
+    options: ConfigRenderOptions = ConfigRenderOptions.defaults())(implicit F: Sync[F], csf: ContextShift[F], writer: Derivation[ConfigWriter[A]]): F[Unit] = {
+    val fileAlreadyExists =
+      F.raiseError(new IllegalArgumentException(s"Cannot save configuration in file '$outputPath' because it already exists"))
+
+    val fileIsDirectory =
+      F.raiseError(new IllegalArgumentException(s"Cannot save configuration in file '$outputPath' because it is a directory"))
+
+    val check = F.suspend {
+      if (!overrideOutputPath && Files.isRegularFile(outputPath)) fileAlreadyExists
+      else if (Files.isDirectory(outputPath)) fileIsDirectory
+      else F.unit
+    }
+
+    blocker.blockOn(check) >> Resource.fromAutoCloseable(F.delay(Files.newOutputStream(outputPath))).use { outputStream =>
```

Comment:
Actually, `newOutputStream` (and  the corresponding `close`) are blocking as well. So we have to do
```scala
Resource.fromAutoCloseable(F.delay(Files.newOutputStream(outputPath)))
  .mapK(blocker.blockOnK[F])
  .use { ... }
```

Reply:
I am not totally sure if those two methods are blocking or not _(I believe not)_. Especially because the official example of `fromAutoClosable` doesn't involve a **Blocker** anywhere.